ROX-27142: clarify automatic upgrades for secured clusters (#85892)
openshift/openshift-docs: configuration/configure-automatic-upgrades.adoc

[id="configure-automatic-upgrades"]
= Configuring automatic upgrades for manifest-installed secured clusters
include::modules/common-attributes.adoc[]
:context: configure-automatic-upgrades

toc::[]

[role="_abstract"]

> If you installed {product-title} by using the manifest installation method, also known as the `roxctl` CLI method, you can automate the upgrade process for each secured cluster. You can also view the upgrade status from the {product-title-short} portal.

Reviewer: > also known as the `roxctl` CLI method

we also call it the "legacy installation method" and I think this is how we refer to it in the UI
![Screenshot 2024-12-06 at 10 32 37](https://github.com/user-attachments/assets/2a4ad38e-f014-4f54-8b49-0e6a11c9a6f6)

Author: Apparently there are 3 names for it, so I will try to work them all in.

> If you installed {product-title} by using the manifest installation method, also known as the `roxctl` CLI method, you can automate the upgrade process for each secured cluster. This process is also called the _legacy installation method_ in the {product-title-short} portal. You can also view the upgrade status from the {product-title-short} portal.

Reviewer: > you can automate the upgrade process for each secured cluster. This process is also called the _legacy installation method_

This is confusing a bit. "This process" does not refer to the upgrade process, but to the installation method. Here is my suggestion:

```suggestion
If you installed {product-title} by using the manifest installation method, also known as the `roxctl` CLI method or the _legacy installation method_, you can automate the upgrade process for each secured cluster. You can also view the upgrade status from the {product-title-short} portal.
```

[NOTE]
====
Automatic upgrades are only available for {product-title-short} systems that were installed by using the manifest installation method. If you installed {product-title-short} by using the Operator, upgrades are controlled by using {olm-first}. If you installed {product-title-short} by using Helm charts, you must use Helm to upgrade.
====

> Automatic upgrades make it easier to stay up-to-date by automating the manual task of upgrading each secured cluster. With automatic upgrades, after you upgrade Central, then Sensor, Collector, and Compliance services in all secured clusters automatically upgrade to the latest version.

Reviewer: > With automatic upgrades, after you upgrade Central, then Sensor, Collector, and Compliance services in all secured clusters automatically upgrade to the latest version.

This part reads weird. The upgrader always upgrades entire secured cluster (Sensor, Collector [which includes Compliance, Collector, and NodeInventory for Scanner V2], Admission Controll) to the same version as Central. So if Central is 4.5 and secured cluster 4.4, then it can be upgraded to 4.5 (but not 4.6, even if it is available).

> Automatic upgrades make it easier to stay up-to-date by automating the manual task of upgrading each secured cluster. If you have automatic upgrades enabled, the upgrader will upgrade the entire secured cluster to the same version as Central.

Reviewer: This block is perfectly fine. If we want to add an important detail, I would suggest this version:

```suggestion
Automatic upgrades make it easier to stay up-to-date by automating the manual task of upgrading each secured cluster. If you have automatic upgrades enabled and the secured cluster is configured for receiving automated upgrades, the upgrader will upgrade the entire secured cluster to the same version as Central.
```

> Automatic upgrades make it easier to stay up-to-date by automating the manual task of upgrading each secured cluster. If you have automatic upgrades enabled, and the secured cluster is configured for receiving automated upgrades, the upgrader will upgrade the entire secured cluster to the same version as Central.

Author: I think it is called "the upgrader" in the code ([example](https://docs.redhat.com/en/documentation/red_hat_advanced_cluster_security_for_kubernetes/4.7/html-single/api_reference/index#UpgradeProgressUpgradeState__v1_clusters_get)) so I am reluctant to change it (plus there are 62 total results on "upgrader" in the docs).

> Automatic upgrades make it easier to stay up-to-date by automating the manual task of upgrading each secured cluster. If you have automatic upgrades enabled, and the secured cluster is configured for receiving automated upgrades, the upgrader will upgrade the entire secured cluster to the same version as Central.

Reviewer 2: It isn't a big deal, and I suspected that this would be the case. Thanks for checking.

> {product-title-short} also enables centralized management of all your secured clusters from within the {product-title-short} portal.

Reviewer 3: > also enables centralized management of all your secured clusters

Is there a clear way to distinguish possible senses of **centralized management**?
* If I understand correctly, **Cluster Status** of **all** makes sense, independent of installation method.
* However, management in an active sense is what these changes aim to limit to less than **all** clusters.
    Someone who knows more than me can comment if even **Delete cluster** makes sense for other installation methods than manifest.

Reviewer 4: "Delete" is a little of a misnomer here, because it just removes the link to the cluster in the ACS database. I don't think it can actually remove the Secured Cluster components from the cluster that is being "deleted from our records". It most definitely does not **destroy** the cluster itself--that still exists, just without any monitoring by ACS.

Author: > > also enables centralized management of all your secured clusters
> 
> Is there a clear way to distinguish possible senses of **centralized management**?
> 
> * If I understand correctly, **Cluster Status** of **all** makes sense, independent of installation method.
> * However, management in an active sense is what these changes aim to limit to less than **all** clusters.
>   Someone who knows more than me can comment if even **Delete cluster** makes sense for other installation methods than manifest.

Hi, I'm finally getting back to this now that 4.7 has shipped. I'm not sure what you mean by the second bullet point.

Reviewer 3: **Delete cluster** deletes information about a secured cluster in Central.

The way that I understand it, either scenario is relevant:
* Customer deletes the cluster itself, therefore the secured cluster no longer exists.
* Customer uninstalls sensor, collector, and so on, from a cluster, therefore no longer secured.

It is definitely relevant for manifest installation, but that is for internal use only.

What I attempted to ask is whether a relevant step for operator or Helm chart installations. If yes, then it seems worthwhile to think whether the content is clear enough, especially cause-and-effect relationship:
* **Delete cluster** in UI that affects data in Central does not uninstall platform from secured cluster.
* Uninstall platform from secured cluster does or does not (question above) delete data from Central.

Author: So I guess I assumed that this section only covered the automatic upgrades and enabling/disabling them. I guess we could add a section elsewhere about managing/deleting clusters by using the UI (I only see it covered under [roxctl cli ](https://85892--ocpdocs-pr.netlify.app/openshift-acs/latest/cli/managing-secured-clusters) in the docs, unless I'm just missing it).
